codegen/templates/http_client.tmpl

```diff
@@ -67,15 +64,15 @@ func (c *{{$clientName}}) {{title .Name}}(ctx context.Context, r *{{.RequestType
 		return nil, err
 	}
 	req.Header.Set("Content-Type", "application/json")
-	return c.Client.Do(req.WithContext(ctx))
+	return c.client.Client.Do(req.WithContext(ctx))
```

Comment:
Good point, I'd rather avoid inheriting http.Client as public API for now. 

naming these things is a bit hard.